toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/AuthenticatorState.kt

```diff
@@ -164,29 +191,74 @@ private class AuthenticatorStateImpl(
+     * given [challenge.requestUrl]. If it can, it will handle the OAuth challenge and return
+     * an [ArcGISAuthenticationChallengeResponse] with an [OAuthUserCredential].
+     * If it cannot, it will handle the ArcGIS token challenge instead.
+     *
+     * @param challenge the [ArcGISAuthenticationChallenge] that requires authentication.
+     * @return an [ArcGISAuthenticationChallengeResponse] with an [ArcGISCredential] or
+     * [ArcGISAuthenticationChallengeResponse.Cancel] if the user cancelled the challenge.
+     * @since 200.8.0
+     */
+    private suspend fun handleOAuthOrTokenChallenge(
+        challenge: ArcGISAuthenticationChallenge
+    ): ArcGISAuthenticationChallengeResponse {
         val oAuthUserConfiguration = oAuthUserConfiguration
-        return if (oAuthUserConfiguration != null && oAuthUserConfiguration.canBeUsedForUrl(challenge.requestUrl)) {
-            val oAuthUserCredential =
-                oAuthUserConfiguration.handleOAuthChallenge {
-                    // A composable observing [pendingOAuthUserSignIn] can launch the OAuth prompt
-                    // when this value changes.
-                    _pendingOAuthUserSignIn.value = it
-                }.also {
-                    // At this point we have suspended until the OAuth workflow is complete, so
-                    // we can get rid of the pending sign in. Composables observing this can know
-                    // to remove the OAuth prompt when this value changes.
-                    _pendingOAuthUserSignIn.value = null
-                }.getOrThrow()
-
-            ArcGISAuthenticationChallengeResponse.ContinueWithCredential(
-                oAuthUserCredential
-            )
+        return if (oAuthUserConfiguration?.canBeUsedForUrl(challenge.requestUrl) == true) {
+            val oAuthUserCredential = oAuthUserConfiguration.handleOAuthChallenge { _pendingOAuthUserSignIn.value = it }
+                .also { _pendingOAuthUserSignIn.value = null }
+                .getOrThrow()
+
+            ArcGISAuthenticationChallengeResponse.ContinueWithCredential(oAuthUserCredential)
         } else {
-            handleArcGISTokenChallenge(challenge)
+            return handleArcGISTokenChallenge(challenge)
         }
     }
 
+    /**
+     * Handles an IAP-based sign in by suspending until the user has completed the sign in or cancelled it.
+     *
+     * @param requestUrl the URL that issued the IAP challenge.
+     * @return an [ArcGISAuthenticationChallengeResponse] with an [IapCredential] or [ArcGISAuthenticationChallengeResponse.Cancel]
+     * if the user cancelled.
+     * @since 200.8.0
+     */
+    private suspend fun handleIapChallenge(requestUrl: String): ArcGISAuthenticationChallengeResponse {
+        val iapConfigurations = iapConfigurations.value
+        val matchingIapConfiguration = iapConfigurations.firstOrNull {
+            it.canBeUsedForUrl(requestUrl)
+        }
```

Comment:
This can be simpliefied to:
```suggestion
        val matchingIapConfiguration = iapConfigurations.value.firstOrNull {
            it.canBeUsedForUrl(requestUrl)
        }
```